### PR TITLE
IGAPP-122: Send user to landing page if current city becomes inactive

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/Navigator.tsx
Comment (LeandraH) [inline native/src/Navigator.tsx:137]: Is this a todo for the future? Then we can maybe add a `TODO` to make it easier to find? Or can it be removed?

File: native/src/Navigator.tsx
Comment (LeandraH) [inline native/src/Navigator.tsx:155]: Thanks, I think that would have been really hard to understand :)

**Review by LeandraH — COMMENTED** [review]
It looks reasonable but I can't get the functionality to work on the emulated iPhone I'm using. I archived Aichach-Friedberg and I can still change the language in there. I also exited the app on the phone and re-opened, I pulled down to refresh, I even restarted it from XCode, nothing has sent me back to the selection screen so far :(

File: native/src/Navigator.tsx
Comment (steffenkleinle) [reply native/src/Navigator.tsx:137]: Its not a todo, more like an instruction if our async storage/app settings schema changes at some point where to put migration routines. But yeah, I think we can remove it.

**Comment by steffenkleinle** [comment]
> It looks reasonable but I can't get the functionality to work on the emulated iPhone I'm using. I archived Aichach-Friedberg and I can still change the language in there. I also exited the app on the phone and re-opened, I pulled down to refresh, I even restarted it from XCode, nothing has sent me back to the selection screen so far :(

To test this, you can just add a `const cities = data.filter(it => it.code !== 'testumgebung')` in the navigator which just simulates the cms not delivering the city with the api anymore.

File: native/src/__tests__/Navigator.spec.tsx
Comment (sarahsporck) [inline native/src/__tests__/Navigator.spec.tsx:185]: findByText and waitForExpect should do the same: https://github.com/testing-library/dom-testing-library/blob/d09b3c25b07486c17603317f4c5a0320c188cbed/src/query-helpers.ts#LL120C1-L120C1 If this is making a difference it's because you are missing thi await

**Review by sarahsporck — APPROVED** [review]
Tested on Android emulator.

**Review by f1sh1918 — APPROVED** [review]
Tested on iOS. Works fine :)

**Comment by steffenkleinle** [comment]
I will recheck this once I am back so pls dont merge :)

File: native/src/__tests__/Navigator.spec.tsx
Comment (steffenkleinle) [reply native/src/__tests__/Navigator.spec.tsx]: Thanks for the hint :+1:

**Comment by steffenkleinle** [comment]
> It looks reasonable but I can't get the functionality to work on the emulated iPhone I'm using. I archived Aichach-Friedberg and I can still change the language in there. I also exited the app on the phone and re-opened, I pulled down to refresh, I even restarted it from XCode, nothing has sent me back to the selection screen so far :(

I could reproduce this (partly). If you don't close the app, the hook in Navigator will never be called and the user is therefore not sent back to the city selection. Instead, I now improved the error displayed if the city does not exist anymore and added the possibility to manually navigate to the city selection (since clearing the city code while for example the categories screen is still opened leads to errors being thrown). On the next app start, the city should automatically be cleared now together with a shown snackbar (if the user did not manually select a new city beforehand).

**Review by LeandraH — APPROVED** [review]
Awesome! Got the two described error messages!

**Pull request merged**
